meshpy/mesh_creation_functions/beam_basic_geometry.py

```diff
@@ -482,3 +483,128 @@ def create_beam_mesh_arc_at_node(
+    mesh,
+    beam_object,
+    material,
+    axis,
+    axis_point,
+    start_point,
+    twist_angle,
+    *,
+    height_helix=None,
+    turns=None,
+    **kwargs
+):
+    """
+    Generate a helical segment starting at a given start point around a
+    predefined axis.
+
+    Args
+    ----
+    mesh: Mesh
+        Mesh that the helical segment will be added to.
+    beam_object: Beam
+        Class of beam that will be used for this line.
+    material: Material
+        Material for this segment.
+    axis: np.array, list
+        Rotation axis of the helix.
+    axis_point: np.array, list
+        Point lying on the rotation axis. Does not need to align with
+        bottom plane of helix.
+    start_point: np.array, list
+        Start point of the helix. Defines the radius.
+    twist_angle: float
+        Twist angle / Pitch of helix between 0 and 2pi.
+    height_helix: float
+        Height of helix. Mutually exclusive with number of turns.
+    turns: float
+        Number of turns. Mutually exclusive with height of helix.
+
+    **kwargs (for all of them look into create_beam_mesh_function)
+    ----
+    n_el: int
+        Number of equally spaced beam elements along the line. Defaults to 1.
+        Mutually exclusive with l_el.
+    l_el: float
+        Desired length of beam elements. Mutually exclusive with n_el.
+        Be aware, that this length might not be achieved, if the elements are
+        warped after they are created.
+
+    Return
+    ----
+    return_set: GeometryName
+        Set with the 'start' and 'end' node of the line. Also a 'line' set
+        with all nodes of the line.
+    """
+
+    if height_helix is None and turns is None:
+        raise ValueError("Either provide height_helix or turns!")
+    elif height_helix is not None and turns is not None:
+        raise ValueError("Only provide height_helix OR turns!")
+
+    # determine radius of helix
+    axis = np.array(axis) / np.linalg.norm(np.array(axis))
+    origin = axis_point + np.dot(
+        np.dot((np.array(start_point) - np.array(axis_point)), axis), axis
+    )
+    start_point_origin_vec = start_point - origin
+    radius = np.linalg.norm(start_point_origin_vec)
+
+    # return line if radius of helix is 0 or twist angle is np.pi/2
+    if np.isclose(radius, 0) or np.isclose(twist_angle, np.pi / 2):
+        if turns:
+            raise ValueError(
+                "Radius of helix is 0 or twist angle is 90 degrees! "
+                + "Height of helix can not be determined through turns! "
+                + "Either switch to height of helix or change radius!"
+            )
+
+        warnings.warn(
+            "Radius of helix is 0 or twist angle is 90 degrees! "
+            + "Simple line geometry is returned!"
+        )
+
+        return create_beam_mesh_line(
+            mesh,
+            beam_object,
+            material,
+            start_point=start_point,
+            end_point=start_point + height_helix * axis,
```

Comment:
To be consistent with the case of "negative" slopes, this should be
```python
end_point=start_point + height_helix * axis * np.sin(twist_angle),
```

Reply:
I am unsure why you would add the `sin` here? Probably you meant `np.sign`. Added in latest commit

Reply:
Since this is only computed for `cos(twist_angle) == 0` you don't need the `sign` here, but I am fine if we add this for clarity.

Reply:
ooh yea you're totally right. removed the sign

Reply:
During further testing on own examples, the signum is actually needed. We also create a straight beam geometry for `radius == 0`, there `twist_angle != np.pi/2` and the geometry is artificially lengthened or shortened

I pushed the fix